src/Discord.Net.WebSocket/Entities/Channels/SocketCategoryChannel.cs

```diff
@@ -15,7 +15,7 @@ namespace Discord.WebSocket
     public class SocketCategoryChannel : SocketGuildChannel, ICategoryChannel
     {
         public override IReadOnlyCollection<SocketGuildUser> Users
-            => Guild.Users.Where(x => x.VoiceChannel?.Id == Id).ToImmutableArray();
+            => throw new NotSupportedException();
```

Comment:
Hm, makes sense. I'll update it.

Reply:
As talked in the dev guild, resolving the permissions could be misleading, since if the users don't have read access to any channels inside, they won't be able to see the category too.

A suggestion would be `Guild.Channels.Where(x => x.CategoryId == Id).SelectMany(x => x.Users).Distinct().ToImmutableArray();` that would list all users that can see at least one channel (and the category consequently) but doesn't mean all of them can see all the channels inside it.